vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/FieldValidationUtil.java

```diff
@@ -31,36 +29,41 @@ private FieldValidationUtil() {
+                            + "validation can only be disabled for a component "
+                            + "when it has been attached to the UI and because "
+                            + "it should be called again once the component is "
+                            + "removed/added, you should call this method from "
+                            + "the onAttach() method of the component.",
+                    component.toString()));
+        }
+        // Wait until the response is being written as the validation state
+        // should not change after that
+        final StateNode componentNode = component.getElement().getNode();
+        componentNode.runWhenAttached(ui -> ui.getInternals().getStateTree()
+                .beforeClientResponse(componentNode,
+                        executionContext -> overrideClientValidation(
+                                component)));
     }
 
-    private static void overrideClientValidation(Component component) {
-        PendingJavaScriptResult javaScriptResult = component.getElement()
-                .executeJs(
-                        "this.validate = function () {return this.checkValidity();}");
+    private static void overrideClientValidation(DateTimePicker component) {
+        StringBuilder expression = new StringBuilder(
+                "this.validate = function () {return this.checkValidity();};");
 
-        javaScriptResult.then(result -> {
-            if (component instanceof HasValidation
-                    && ((HasValidation) component).isInvalid()) {
-                /*
-                 * By default, the invalid flag is always false when a component
-                 * is created. However, if the component is populated and
-                 * validated in the same HTTP request, the server side state may
-                 * have changed before the JavaScript disabling client side
-                 * validation was properly executed. This can sometimes lead to
-                 * a situation where the client side thinks the value is valid
-                 * (before client side validation was disabled) and the server
-                 * side thinks the value is invalid. This will lead to strange
-                 * behavior until the two states are synchronized again. To
-                 * avoid this, we will explicitly change the client side value
-                 * if the server side is invalid.
-                 */
-                component.getElement().executeJs("this.invalid = true");
-            }
-        });
+        if (component.isInvalid()) {
+            /*
+             * By default the invalid flag is set to false. Workaround the case
+             * where the client side validation overrides the invalid state
+             * before the validation function itself is overridden above.
+             */
+            expression.append("this.invalid = true;");
+            expression.append("this.validate();");
```

Comment:
Are these calls to `this.validate()` needed? They were not before this change.

Reply:
I added that to basically make sure the invalid state is applied correctly in the component, regardless of it is needed or not.
I have no idea how redundant it is as I don't know the web component(s) that well. This is basically "just in case" setting the attribute would not be enough in some certain case or for some component.

I can remove the part if it should never be needed or add a comment that clarifies that this is "just in case and maybe reduntant" ?

Reply:
As the patched `validate` only returns `checkValidity` value and this method doesn't produce any side effect, my understanding is that it's not needed.

Reply:
Done